app/backend/nvhttp.cpp

```diff
@@ -57,8 +57,13 @@ NvHTTP::getCurrentGame(QString serverInfo)
     // GFE 2.8 started keeping currentgame set to the last game played. As a result, it no longer
     // has the semantics that its name would indicate. To contain the effects of this change as much
     // as possible, we'll force the current game to zero if the server isn't in a streaming session.
+    //
+    // However, current game info must be available also in other states than just _SERVER_BUSY as
+    // it is required for quitting currently running app. Quitting app occurs at end of stream if
+    // configured so. At that point the server state may be in some other state than _SERVER_BUSY
+    // for a short while, but that must not prevent quitting of the app.
     QString serverState = getXmlString(serverInfo, "state");
-    if (serverState != nullptr && serverState.endsWith("_SERVER_BUSY"))
+    if (serverState != nullptr && !serverState.endsWith("_SERVER_AVAILABLE"))
```

Comment:
There were several versions of GFE that had a bug where they left the server in state `_SERVER_QUITTING` even after the game was closed, so we need to treat `_SERVER_QUITTING` as if it was `_SERVER_AVAILABLE`.

What state were you seeing the server in when you were seeing issues quitting?

Reply:
This at least happens when starting app from UI. The last time when NvComputer updates is at start of the stream where server state is  MJOLNIR_STATE_SERVER_LAUNCHING, which resets the app id to zero. When the session window is closed, quitApp() gets called just before NvComputer would update, which does nothing as app id is still zero.